File: POTD/Difference Array/Cpp/1674. Minimum Moves to Make Array Complementary.cpp
class Solution {
public:
    int minMoves(vector<int>& nums, int limit) {
        int n=nums.size();
        vector<int> diff(2*limit+2,0);

        for(int i=0;i<n/2;i++){
            int minTargetVal=min(nums[i],nums[n-i-1])+1;
            int maxTargetVal=max(nums[i],nums[n-i-1])+limit;

            int sum=nums[i]+nums[n-i-1];
            diff[2]+=2;
            diff[minTargetVal]-=1;
            diff[maxTargetVal+1]+=1;
            diff[2*limit+1]-=2;
            diff[sum]-=1;
            diff[sum+1]+=1;
        }

        for(int i=3;i<=(2*limit+1);i++){
            diff[i]=diff[i-1]+diff[i];
        }
        int minMoves=INT_MAX;
        for(int i=2;i<=2*limit;i++){
            minMoves=min(minMoves,diff[i]);
        }

        return minMoves;
    }

};

File: POTD/Difference Array/Python/1674. Minimum Moves to Make Array Complementary.py
class Solution:
    def minMoves(self, nums: List[int], limit: int) -> int:
        n=len(nums)
        diff=[0]*(2*limit+2)

        for i in range(0,n//2):
            sum=nums[i]+nums[n-i-1]
            low=min(nums[i],nums[n-i-1])+1
            high=max(nums[i],nums[n-i-1])+limit

            diff[2]+=2
            diff[2*limit+1]-=2

            diff[low]-=1
            diff[high+1]+=1

            diff[sum]-=1
            diff[sum+1]+=1
        

        for i in range(3,2*limit+2):
            diff[i]=diff[i-1]+diff[i]
        

        minMoves=float('inf')
        for i in range(2,2*limit+1):
            minMoves=min(minMoves,diff[i])
        
        return minMoves
